### ISSUE TITLE
Calling getCSSText returns [object Object] or [object Undefined] with Remix

### ISSUE THREAD

**Issue by joshuaellis** [modulz/stitches#908]
Whilst I appreciate there's no formal documentation as of yet, I followed this [gist](https://gist.github.com/jjenzz/b67c8bd263bbf0f4e39c5fca55db93ee) as suggested by a fair few folk on the internet.

My attempt can be seen here: https://github.com/joshuaellis/remix-stitches

However, when you run that demo, you can see the SSR side of the styling works correctly, the `Hello World` is the specified red from the theme created by `createStitches`. When the client side hydrates, `getCssText` does not call the internal `toString` property you had written, it returns the native `object.toString` function, which therefore returns `[object Object]` or in some cases `[object Undefined]`.

What I believe to be happening during client hydration is `groupSheet` is becoming this object:

https://github.com/modulz/stitches/blob/c48a6532db350093dbd00fa2bace29930c53ba68/packages/core/src/sheet.js#L86

which evidently does not have a `toString` function. Patching this function does return the `theme` stylesheets, with all the colors applied to `:root` but then the text is still not "red" because the styles from the components and their class names have not been collected and put into the `cssRules` object.

Any help / guidance would be deeply appreciated!

**Comment by JustinVoitel** [comment]
Same applies for SvelteKit applications!

I tried renaming the internal `toString` method of the sheet object to something else and it fixed the error in my case, so pretty trivial change imo.

**Comment by jckw** [comment]
Interestingly, I'm finding that if I don't call `getCssText` in root.tsx or entry.server.tsx, this issue doesn't occur.

There are then other issues though:

1. the return value of `getCssText` differ on server and client because of new lines (which throws a console error - otherwise minor)
2. classnames generated by the `css()` and `styled()` funcs are not picked up on either the client or server output of `getCssText()`


---
Other potentially related things I've noticed:

3. calling `getCssText()` before the call to `renderToString` in entry.server.tsx gives a different output to calling it afterwards, namely missing css and styled generated classnames
4. calling `getCssText()` in the Document component also misses the classnames generated by css and styled (which can probably be explained via (3))

What confuses me is how the Next JS example works so simply, but Remix is more complex. I suppose I'd need to know more about the inner workings of both libraries to be able to work out why. 

I hope this helps!

**Comment by joshuaellis** [comment]
My understanding of 3. & 4. based on reading threads and messages in the Remix discord alongside my understanding of react, is that the tree needs to be rendered so that the `styled` components within the tree have their classnames and styles  generated (thus adding to the CSS sheet). I believe (and this is tenuous) that this is why some CSS-in-JS libraries do a "double render".

Removing the `Scripts` component from the `remix` `root.tsx` styles everything correctly (if you follow the `entry.server.tsx` from the gist) because styles are correctly collected after `renderToString`. But the hydration breaks the styling because of the bug described in the original Issue.

**Comment by joshuaellis** [comment]
Following up on this with some more investigation, reviewing the compiled version, the `isSheetAccessible` function:

```js
const isSheetAccessible = (/** @type {CSSStyleSheet} */ sheet) => {
	if (sheet.href && !sheet.href.startsWith(location.origin)) {
		return false
	}

	try {
		sheet.cssRules
		return true
	} catch (e) {
		return false
	}
}
```

which is used like this:


```js
if (!isSheetAccessible(sheet)) continue
```

becomes something like this:

```js
	j = (e) => {
		if (e.href && !e.href.startsWith(location.origin)) return !1
		try {
			return e.cssRules, !0
		} catch (e) {
			return !1
		}
	},
```

and is used like this:

```js
if (j(e)) {
```

Which I think is confusing because `isSheetAccessible` should be returning true (which then is false because of the checked) this is because the stylesheet has no URL but has CSSRules. But in J, it returns `true` as expected but the function when called will pass. It looks like something strange is happening in the build process? Therefore modifying the function check to be:

```js
if (!j(e)) {
```

Restores the the functionality of getCssText giving us styled content. However, the issue still persists that the `styled` components are not collected and rendered.

***permalink to code reference***
https://github.com/modulz/stitches/blob/c48a6532db350093dbd00fa2bace29930c53ba68/packages/core/src/sheet.js#L18

**Comment by joshuaellis** [comment]
The `threejs` dev tools extension broke stitches. I'm not sure what to suggest on this... but i'll leave this here for now.

**Comment by luixo** [comment]
Adding `toString` function defined in [no-hydratable-stylesheet branch](https://github.com/modulz/stitches/blob/cb414be79693c783b60923aeda939df6006a0020/packages/core/src/sheet.js#L118-L143) to [groupSheet definition](https://github.com/modulz/stitches/blob/canary/packages/core/src/sheet.js#L86) helped me in the same case as given here.

Had to use [patch-package](https://github.com/ds300/patch-package) though.

**Comment by hadihallak** [comment]
Fixed in 1.2.7
